### PR TITLE
Finish Subject by trainee

### PR DESCRIPTION
**Pull request by Rownaknira**

### CODE REVIEW

File: app/views/courses_subjects/edit.html.erb [hunk 1]
```diff
@@ -6,12 +6,21 @@
...
-      <%= f.label(:title, t ".check" ) %><br>
+      <%= f.label(:title, t(".check")) %><br>
       <%= f.collection_check_boxes(:task_ids, @courses_subject.tasks, :id, :name) do |b| %>
-        <%= b.label(class: "checkbox-inline") {b.check_box + b.text} %>
+        <%= b.label(class: "checkbox-inline") { b.check_box + b.text } %>
       <% end %>
     </div>
-    <center><%= f.submit t(".update"), class: "btn btn-primary" %></center>
+    <div>
+      <%= check_box_tag :users_subject_ids, @courses_subject.id,"", :name => 'user[courses_subject_ids][]', :style => 'visibility: hidden'%>
```
Comment (zenith19) [inline app/views/courses_subjects/edit.html.erb]: check 80 char

**Comment by zenith19** [comment]
Ok

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -10,10 +10,18 @@ def show
   end
 
   def update
-    if @user.update_attributes user_params
-      redirect_to course_path
-    else
-      render "courses_subjects/edit"
+    if params[:commit] == "Update"
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: define constant

File: app/views/courses_subjects/edit.html.erb [hunk 3]
```diff
@@ -6,12 +6,22 @@
...
+      <%= f.label(:title, t(".check")) %><br>
       <%= f.collection_check_boxes(:task_ids, @courses_subject.tasks, :id, :name) do |b| %>
-        <%= b.label(class: "checkbox-inline") {b.check_box + b.text} %>
+        <%= b.label(class: "checkbox-inline") { b.check_box + b.text } %>
       <% end %>
     </div>
-    <center><%= f.submit t(".update"), class: "btn btn-primary" %></center>
+    <div>
+      <%= check_box_tag :users_subject_ids, @courses_subject.id, "",
+        :name => 'user[courses_subject_ids][]', :style => 'visibility: hidden'%>
```
Comment (chuanhtuan) [inline app/views/courses_subjects/edit.html.erb]: =>

File: app/views/courses_subjects/edit.html.erb [hunk 4]
```diff
@@ -6,12 +6,22 @@
...
-      <%= f.label(:title, t ".check" ) %><br>
+      <%= f.label(:title, t(".check")) %><br>
       <%= f.collection_check_boxes(:task_ids, @courses_subject.tasks, :id, :name) do |b| %>
-        <%= b.label(class: "checkbox-inline") {b.check_box + b.text} %>
+        <%= b.label(class: "checkbox-inline") { b.check_box + b.text } %>
       <% end %>
     </div>
-    <center><%= f.submit t(".update"), class: "btn btn-primary" %></center>
+    <div>
+      <%= check_box_tag :users_subject_ids, @courses_subject.id, "",
```
Comment (chuanhtuan) [inline app/views/courses_subjects/edit.html.erb]: collection_check_boxes

Comment (chuanhtuan) [inline app/views/courses_subjects/edit.html.erb]: remove
no need

File: app/views/courses_subjects/edit.html.erb [hunk 5]
```diff
@@ -6,12 +6,22 @@
...
+        :name => 'user[courses_subject_ids][]', :style => 'visibility: hidden'%>
+    </div>
+    <%= f.hidden_field :user_id, value: current_user.id %>
```
Comment (chuanhtuan) [inline app/views/courses_subjects/edit.html.erb]: remove

File: app/views/courses_subjects/edit.html.erb [hunk 6]
```diff
@@ -6,12 +6,22 @@
...
+    <% if @courses_subject.users_subjects.find_by("user_id = ?", current_user.id)== nil %>
```
Comment (chuanhtuan) [inline app/views/courses_subjects/edit.html.erb]: remove
no need to check
only can update when course is not started

File: app/views/courses_subjects/edit.html.erb (hunk 4 above)
Comment (Rownaknira) [inline app/views/courses_subjects/edit.html.erb]: I use users_subject_ids to insert a row in UsersSubject table. Whenever a trainee wants to finish a subject, he will click the finish button and a corresponding row will be inserted into UsersSubject table

File: app/views/courses_subjects/edit.html.erb [hunk 7]
```diff
@@ -6,12 +6,21 @@
...
-        <%= b.label(class: "checkbox-inline") {b.check_box + b.text} %>
+        <%= b.label(class: "checkbox-inline") { b.check_box + b.text } %>
       <% end %>
     </div>
-    <center><%= f.submit t(".update"), class: "btn btn-primary" %></center>
+    <div>
+      <%= check_box_tag :users_subject_ids, @courses_subject.id, "",
+        :name => 'user[courses_subject_ids][]', :style => 'visibility: hidden'%>
+    </div>
+    <% if @courses_subject.users_subjects.find_by("user_id = ?", current_user.id)== nil %>
```
Comment (Rownaknira) [inline app/views/courses_subjects/edit.html.erb]: If a trainee finishes a subject, he will not get finish and update button

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -10,10 +13,18 @@ def show
...
-      redirect_to course_path
-    else
-      render "courses_subjects/edit"
+    if params[:commit] == UsersController::ACTION_TYPES_UPDATE
+      if @user.update_attributes user_params
+        redirect_to course_path
+      else
+        render "courses_subjects/edit"
+      end
+    elsif params[:commit] == UsersController::ACTION_TYPES_FINISH
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: i think, you should write in user_subjects_controller

**Pull request merged**